lms/djangoapps/courseware/access.py

```diff
@@ -304,6 +358,51 @@ def can_view_courseware_with_prerequisites():  # pylint: disable=invalid-name
     return _dispatch(checkers, action, user, course)
 
 
+COURSE_OVERVIEW_SUPPORTED_ACTIONS = [  # pylint: disable=invalid-name
+    'load',
+    'load_mobile',
+    'view_courseware_with_prerequisites'
+]
+
+
+def _has_access_course_overview(user, action, course_overview):
+    """
+    Check if user has access to a course overview.
+
+    Arguments:
+        user (User): the user whose course access we are checking.
+        action (str): the action the user is trying to perform.
+            See COURSE_OVERVIEW_SUPPORTED_ACCESS_TYPES for valid values.
+        course_overview (CourseOverview): overview of the course in question.
+    """
+    def can_load():
+        """
+        Can this user load this course?
+
+        NOTE: this is not checking whether user is actually enrolled in the course.
+        """
+        return (
+            not course_overview.visible_to_staff_only
+            and _can_access_descriptor_with_start_date(user, course_overview, course_overview.id)
+        ) or _has_staff_access_to_descriptor(user, course_overview, course_overview.id)
+
+    checkers = {
+        'load': can_load,
+        'load_mobile': lambda: can_load() and _can_load_course_on_mobile(user, course_overview),
+        'view_courseware_with_prerequisites':
+            lambda: _can_view_courseware_with_prerequisites(user, course_overview),
+    }
+
+    # Make sure checkers and COURSE_OVERVIEW_SUPPORTED_ACTIONS stay in sync.
+    actions = checkers.keys()
+    assert (
+        all(action in COURSE_OVERVIEW_SUPPORTED_ACTIONS for action in actions) and
+        len(COURSE_OVERVIEW_SUPPORTED_ACTIONS) == len(actions)
```

Comment:
Yea, I'm at loss of any pretty way of checking that `checkers` and `COURSE_OVERVIEW_SUPPORTED_ACTIONS` stay in sync that doesn't involve making `can_load` a module-level function.